### PR TITLE
Qinwen/mlperf 4 merge eval

### PR DESCRIPTION
**Pull request by suexu1025**
* Add multi-host eval
* reference to
./xpk/v5p-32_eval.sh
* remove `image_id	id	caption` (first line) in dataset/coco/val2014_30k.tsv to avoid dummy read.

### CODE REVIEW

File: src/maxdiffusion/generate.py [hunk 1]
```diff
@@ -83,8 +89,9 @@ def get_unet_inputs(rng, config, batch_size, pipeline, params, prompt_ids, negat
     vae_scale_factor = 2 ** (len(pipeline.vae.config.block_out_channels) - 1)
     guidance_scale = config.guidance_scale
     num_inference_steps = config.num_inference_steps
-
+    print(prompt_ids.shape)
```
Comment (entrpn) [inline src/maxdiffusion/generate.py]: remove print statements if they're just being used for debugging.

File: src/maxdiffusion/generate.py [hunk 2]
```diff
@@ -146,8 +153,8 @@ def run(config):
     devices_array = create_device_mesh(config)
     mesh = Mesh(devices_array, config.mesh_axes)
 
-    batch_size = jax.device_count() * config.per_device_batch_size
-    assert 30_000 % batch_size == 0, f"Coco dataset must be evenly divisible by batch size : {batch_size}"
+    batch_size = jax.local_device_count() * config.per_device_batch_size
+    #assert 30_000 % batch_size == 0, f"Coco dataset must be evenly divisible by batch size : {batch_size}"
```
Comment (entrpn) [inline src/maxdiffusion/generate.py]: remove commented out code.

**Comment by entrpn** [comment]
@suexu1025 did you verify that this will also work when running training with `eval_at_checkpoint` set to True?

**Comment by suexu1025** [comment]
> @suexu1025 did you verify that this will also work when running training with `eval_at_checkpoint` set to True?

Yes, just fixed the issue and verified.

**Review by entrpn — APPROVED** [review]

**Pull request merged**